service/history/shardContext.go

```diff
@@ -585,6 +589,19 @@ func (s *shardContextImpl) updateShardInfoLocked() error {
 	}
 	updatedShardInfo := copyShardInfo(s.shardInfo)
 
+	if s.messageProducer != nil {
```

Comment:
Can we only create sync task on idle scenario?

Reply:
#956